src/main/java/pl/javastart/task/Room.java

```diff
@@ -0,0 +1,51 @@
+package pl.javastart.task;
+
+public class Room {
+    private double area;
+    private double temperature;
+    private boolean isAirConditioned;
+    private double lowerLimitTemperature;
+
+    public double getArea() {
+        return area;
+    }
+
+    public double getTemperature() {
+        return temperature;
+    }
+
+    public boolean isAirConditioned() {
+        return isAirConditioned;
+    }
+
+    public double getLowerLimitTemperature() {
+        return lowerLimitTemperature;
+    }
+
+    public Room(double area, double temperature, boolean isAirConditioned, double lowerLimitTemperature) {
+        this.area = area;
+        this.temperature = temperature;
+        this.isAirConditioned = isAirConditioned;
+        this.lowerLimitTemperature = lowerLimitTemperature;
+    }
+
+    private void checkTemperature() {
+        if (temperature < lowerLimitTemperature && isAirConditioned) {
+            temperature = lowerLimitTemperature;
+        }
+    }
+
+    boolean reduceTemperature() {
+        checkTemperature();
+        if (isAirConditioned) {
+            if ((temperature - 1) <= lowerLimitTemperature) {
+                temperature = lowerLimitTemperature;
+                return false;
```

Comment:
Jeśli było 20,5, a min to 20,0, to zwrócisz `false`, mimo że temperatura została obniżona.

Reply:
"Metoda powinna zwracać true, jeżeli udało się obniżyć temperaturę, **lub false jeżeli w pokoju nie ma klimatyzatora lub osiągnięto temperaturę graniczną**." - zrozumiałem to tak, że osiągamy temperaturę graniczną np. obniżając o brakujące 0,5C więc zwracamy false.

Reply:
ok, ma to sens